docs/weave/welcome.md

```diff
@@ -0,0 +1,120 @@
+| [Batch](https://github.com/iov-one/weave/tree/master/x/batch)                    | Used for combining multiple transactions into one atomic operation. A powerful example is in creating single-chain swaps.                                            |
+| [Validators](https://github.com/iov-one/weave/tree/master/x/batch)               | Used in a PoA context to update the validator set using either multisig or the on-chain elections module                                                             |
+| [Username](https://github.com/iov-one/weave/tree/master/cmd/bnsd/x/nft/username) | Example nft used by bnsd. Maps usernames to multiple chain addresses, including reverse lookups                                                                      |
+| [MessageFee](https://github.com/iov-one/weave/tree/master/x/msgfee)              | Validator-subjective minimum fee module, designed as an anti-spam measure.                                                                                           |
+| [Utils](https://github.com/iov-one/weave/tree/master/x/utils)                    | A range of utility functions such as KeyTagger which is designed to enable subscriptions to database.                                                                |
+
+### In Progress
+
+Light client proofs, custom token issuance and support for IBC (Inter Blockchain Communication) are currently being designed.
+
+## Basic Blockchain Terminology
+
+If you are new to blockchains (or Tendermint), this is a crash course in just enough theory to follow the rest of the setup. [Read all](weave/basics/01-blockchain.md)
+
+### Immutable Event Log
+
+If you are coming from working on typical databases, you can think of the blockchain as an immutable [transaction log](https://en.wikipedia.org/wiki/Transaction_log) . If you have worked with [Event Sourcing](https://martinfowler.com/eaaDev/EventSourcing.html) you can consider a block as a set of events that can always be replayed to create a [materialized view](https://docs.microsoft.com/en-us/azure/architecture/patterns/materialized-view) . Maybe you have a more theoretical background and recognize that a blockchain is a fault tolerant form of [state machine replication](https://en.wikipedia.org/wiki/State_machine_replication#Ordering_Inputs) . [Read more](weave/basics/01-blockchain.md#immutable-event-log)
+
+### General Purpose Computer
+
+Ethereum pioneered the second generation of blockchain, where they realized that we didn't have to limit ourselves to handling payments, but actually have a general purpose state machine. [Read more](weave/basics/01-blockchain.md#general-purpose-computer)
+
+### Next Generation
+
+Since that time, many groups are working on "next generation" solutions that take the learnings of Ethereum and attempt to build a highly scalable and secure blockchain that can run general purpose programs. [Read more](weave/basics/01-blockchain.md#next-generation)
```

Comment:
"Since that time, many groups are working on "next generation" solutions that take the learnings of Ethereum"

Suggest

"Since that time, many groups are working on "next generation" solutions that build on the experience of Ethereum"